packages/framework/src/Component/AddSentryContextSubscriber.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Shopsys\FrameworkBundle\Component;
+
+use App\Environment;
+use RedisException;
+use Sentry\State\Scope;
+use Shopsys\FrameworkBundle\Component\Domain\Domain;
+use Shopsys\FrameworkBundle\Component\Domain\Exception\NoDomainSelectedException;
+use Shopsys\FrameworkBundle\Component\Localization\DisplayTimeZoneProviderInterface;
+use Shopsys\FrameworkBundle\Component\Maintenance\MaintenanceModeSubscriber;
+use Shopsys\FrameworkBundle\Component\Redis\RedisClientFacade;
+use Shopsys\FrameworkBundle\Model\Customer\User\CurrentCustomerUser;
+use Shopsys\FrameworkBundle\ShopsysFrameworkBundle;
+use Symfony\Component\Console\ConsoleEvents;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\Filesystem\Exception\FileNotFoundException;
+use Symfony\Component\HttpKernel\KernelEvents;
+use function Sentry\configureScope;
+
+class AddSentryContextSubscriber implements EventSubscriberInterface
+{
+    /**
+     * @param \Shopsys\FrameworkBundle\Component\Redis\RedisClientFacade $redisClientFacade
+     * @param \Shopsys\FrameworkBundle\Component\Domain\Domain $domain
+     * @param \Shopsys\FrameworkBundle\Component\Localization\DisplayTimeZoneProviderInterface $displayTimeZoneProvider
+     */
+    public function __construct(
+        protected readonly RedisClientFacade $redisClientFacade,
+        protected readonly Domain $domain,
+        protected readonly DisplayTimeZoneProviderInterface $displayTimeZoneProvider,
+        protected readonly CurrentCustomerUser $currentCustomerUser,
+    ) {
+    }
+
+    public function setSentryContext(): void
+    {
+        $context = [
+            'environment' => class_exists('App\Environment') ? Environment::getEnvironment() : '',
+            'version' => ShopsysFrameworkBundle::VERSION,
+            'systemTimeZone' => date_default_timezone_get(),
+            'userId' => $this->currentCustomerUser->findCurrentCustomerUser()?->getId(),
+        ];
+
+        try {
+            $context['maintenance-page'] = $this->redisClientFacade->contains(MaintenanceModeSubscriber::MAINTENANCE_KEY);
```

Comment:
this adds several requests to the Redis to check if the maintenance page is enabled. 
It would be nice to have a provider that caches the first response and keep the maintenance state in memory

Reply:
I have introduced MaintenanceModeFacade for this